### PR TITLE
Open source docker build

### PR DESCRIPTION
**Pull request by max-jacobs**
Updating build to only produce one, versioned image per application and not push. Updating Jenkinsfile to push images on deploys and tag as latest on release. This presumes that snapshot/release images will be in the same repo.

Still required:
- `eg-oss-ci` user
- `eg-oss-settings`
- `docker.hub` property in `eg-oss-parent`
- Test that `DOCKER_ORG = readMavenPom().properties['docker.org']` works as expected

### CODE REVIEW

File: beekeeper-package/pom.xml [hunk 1]
```diff
@@ -17,28 +17,9 @@
...
-          <value>true</value>
-        </property>
-      </activation>
-      <properties>
-        <docker.repo>${artifactory.docker.release.repository.id}.${artifactory.base}</docker.repo>
-      </properties>
-    </profile>
-  </profiles>
+  <properties>
+    <docker.org></docker.org>
```
Comment (AnanaMJ) [inline beekeeper-package/pom.xml]: Should this be something other than null?

File: Jenkinsfile [hunk 2]
```diff
@@ -20,15 +22,26 @@ pipeline {
         echo 'Checking out project...'
         checkout scm
         echo 'Building...'
-        withMaven(jdk: 'OpenJDK11', maven: 'Maven3.6', mavenSettingsConfig: 'hcomdata-artifactory-maven-settings') {
-          sh 'mvn clean deploy jacoco:report checkstyle:checkstyle'
+        withMaven(jdk: 'OpenJDK11', maven: 'Maven3.6', mavenSettingsConfig: 'hcomdata-oss-settings') {
```
Comment (AnanaMJ) [inline Jenkinsfile]: Is `hcomdata-oss-settings` something already created in Jenkins?

And should we make one for eg-oss?

File: beekeeper-package/pom.xml (hunk 1 above)
Comment (max-jacobs) [reply beekeeper-package/pom.xml]: Removed from here, this will be in eg-oss-parent

File: Jenkinsfile [hunk 3]
```diff
@@ -9,9 +9,11 @@ pipeline {
   }
 
   environment {
-    GIT = credentials('s-hcom-jenkins-rbac')
+    GIT = credentials('s-hcom-ci')
     GIT_USERNAME = "${env.GIT_USR}"
     GIT_PASSWORD = "${env.GIT_PSW}"
+
+    DOCKER_ORG = readMavenPom().getProperties().getProperty('docker.org')
```
Comment (AnanaMJ) [inline Jenkinsfile]: You're adding `docker.org` to the parent pom, right?

File: Jenkinsfile [hunk 4]
```diff
@@ -48,13 +61,18 @@ pipeline {
...
+        withMaven(jdk: 'OpenJDK11', maven: 'Maven3.6', mavenSettingsConfig: 'hcomdata-oss-settings') {
           sh """mvn --batch-mode release:prepare release:perform \
                   -Dresume=false \
                   -DreleaseVersion=${RELEASE_VERSION} \
                   -DdevelopmentVersion=${DEVELOPMENT_VERSION} \
                   -DautoVersionSubmodules=true"""
         }
+        echo 'Pushing images...'
+        docker tag $DOCKER_ORG/beekeeper-cleanup:${RELEASE_VERSION} $DOCKER_ORG/beekeeper-cleanup:latest
+        docker tag $DOCKER_ORG/beekeeper-path-scheduler-apiary:${RELEASE_VERSION} $DOCKER_ORG/beekeeper-cleanup:latest
```
Comment (AnanaMJ) [inline Jenkinsfile]: it should be `$DOCKER_ORG/beekeeper-path-scheduler-apiary:latest`

File: beekeeper-package/pom.xml (hunk 1 above)
Comment (AnanaMJ) [reply beekeeper-package/pom.xml]: Ohhh okay :)

File: Jenkinsfile (hunk 2 above)
Comment (max-jacobs) [reply Jenkinsfile]: We have a fresh project there, so its empty, but I think you're right we'll need to use different settings for eg open source. Good spot, thanks

**Comment by max-jacobs** [comment]
Closing this PR to do some testing in Jenkins

**Pull request closed without merging**